core/sdk-core/src/main/java/software/amazon/awssdk/core/FileTransformerConfiguration.java

```diff
@@ -0,0 +1,155 @@
+     */
+    public Optional<FileWriteOption> fileModificationOption() {
+        return Optional.ofNullable(fileModificationOption);
+    }
+
+    /**
+     * Whether to delete the file in the event of a failure.
+     */
+    public Optional<Boolean> deleteOnFailure() {
+        return Optional.ofNullable(deleteOnFailure);
+    }
+
+    /**
+     * Create a {@link Builder}, used to create a {@link FileTransformerConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder().fileWriteOption(fileModificationOption)
+                                   .deleteOnFailure(deleteOnFailure);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FileTransformerConfiguration that = (FileTransformerConfiguration) o;
+
+        if (fileModificationOption != that.fileModificationOption) {
+            return false;
+        }
+        return Objects.equals(deleteOnFailure, that.deleteOnFailure);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = fileModificationOption != null ? fileModificationOption.hashCode() : 0;
+        result = 31 * result + (deleteOnFailure != null ? deleteOnFailure.hashCode() : 0);
+        return result;
+    }
+
+    public enum FileWriteOption {
+        /**
+         * Always create a new file. If the file already exists, {@link FileAlreadyExistsException} will be thrown.
+         * In the event of an error, the SDK will attempt to delete the file (whatever has been written to it so far).
+         */
+        CREATE_NEW,
+
+        /**
+         * Create a new file if it doesn't exist, otherwise replace the existing file.
+         * <p>
+         * In the event of an error, the SDK will NOT attempt to delete the file.
+         */
+        CREATE_OR_REPLACE_EXISTING,
+
+        /**
+         * Create a new file if it doesn't exist, otherwise append the existing file.
+         * <p>
+         * In the event of an error, the SDK will NOT attempt to delete the file.
+         */
+        CREATE_OR_APPEND_EXISTING
+    }
+
+    public interface Builder extends CopyableBuilder<Builder, FileTransformerConfiguration> {
+
+        /**
+         * Configures the file write option
+         *
+         * @param fileWriteOption the file write option
+         * @return This object for method chaining.
+         */
+        Builder fileWriteOption(FileWriteOption fileWriteOption);
+
+        /**
+         * Configures whether the file should be deleted in the event of an error
+         *
+         * @param deleteOnFailure whether to delete the file upon failure
+         * @return This object for method chaining.
+         */
+        Builder deleteOnFailure(Boolean deleteOnFailure);
```

Comment:
Maybe `failureBehavior(FailureBehavior.DELETE)` and `failureBehavior(FailureBehavior.LEAVE)` to leave room for something like `failureBehavior(FailureBehavior.REVERT)` in the future?

Reply:
Good idea!